x/appchain/common/types/params.go

```diff
@@ -0,0 +1,148 @@
+package types
+
+import (
+	fmt "fmt"
+	"time"
+
+	ibchost "github.com/cosmos/ibc-go/v7/modules/core/24-host"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+const (
+	// CoordinatorFeePoolAddrStr is from the coordinator params.
+	// DistributionTransmissionChannel is currently blank.
+	// RewardDenom should be fetched from the add-avs tx.
+	// IBCTimeoutPeriod is from the coordinator params.
+
+	// DefaultSubscriberUnbondingPeriod is the default unbonding period for a subscriber chain.
+	// There is no particular rationale to choose this number, since an
+	// operator can opt-in to or opt-out from all AVSs independently.
+	DefaultSubscriberUnbondingPeriod = 24 * 7 * time.Hour
+	// DefaultBlocksPerDistributionTransmission is the default number of blocks after which a reward distribution is
+	// transmitted to the coordinator. TODO: This can be replaced with epochs.
+	DefaultBlocksPerDistributionTransmission = 1000
+	// DefaultSubscriberRedistributionFraction is the default percentage of rewards that are distributed
+	// to the coordinator.
+	DefaultSubscriberRedistributionFraction = "0.1"
+	// DefaultTransferTimeoutPeriod is the default timeout period for IBC transfers.
+	DefaultTransferTimeoutPeriod = time.Hour
+)
+
+// NewSubscriberParams returns a new subscriber params with the given values.
+func NewSubscriberParams(
+	coordinatorFeePoolAddrStr string,
+	distributionTransmissionChannel string,
+	blocksPerDistributionTransmission int64,
+	subscriberRedistributionFraction string,
+	rewardDenom string,
+	ibcTimeoutPeriod time.Duration,
+	transferTimeoutPeriod time.Duration,
+	unbondingPeriod time.Duration,
+	historicalEntries int64,
+	slashFractionDowntime string,
+	downtimeJailDuration time.Duration,
+	slashFractionDoubleSign string,
+) SubscriberParams {
+	return SubscriberParams{
```

Comment:
The 

> NewSubscriberParams

 function accepts a large number of parameters, and it may be considered to use the Builder pattern or the option pattern to reduce the number of parameters.